aep/general/0136/aep.md.j2

```diff
@@ -1,45 +1,50 @@
+expressed via standard methods; prefer standard methods if possible, due to
+their consistent semantics. (Of course, this only applies if the functionality
+in question actually conforms to the normal semantics; it is _not_ a good idea
+to contort things to endeavor to make the standard methods "sort of work".)
 
-While custom operations vary widely in how they are designed, many principles
+While custom methods vary widely in how they are designed, many principles
 apply consistently:
 
+- The HTTP method for custom methods **should** usually be `POST`, unless the
+  custom method maps more strongly to another HTTP verb.
+  - Custom methods that serve as an alternative to or variant of a standard
+    method **should** generally follow all applicable guidance for that method.
+    For example, a protobuf custom method that is an alternative to `ListBooks`
+    **should** have a response where the first field looks like
+    `repeated Book books = 1;`.
+  - Custom methods that serve as an alternative to get or list methods (such as
+    `Search`) **should** use `GET`, and require no request body. These methods
+    **must** be idempotent and have no state changes or side effects (they
+    should be safe as defined in [RFC 7231][]).
+  - Custom methods **should not** use `PATCH` or `DELETE`.
+- The HTTP URI **must** use a `:` character followed by the custom verb
+  (`:archive` in the above example), and the verb in the URI **must** match the
+  verb in the name of the RPC.
+  - If word separation is required, `camelCase` **must** be used.
+
 {% tab proto %}
 
 {% sample 'library.proto', 'rpc ArchiveBook' %}
 
 - The name of the RPC **should** be a verb followed by a noun.
   - The name **must not** contain prepositions ("for", "with", etc.).
-- The HTTP method for custom operations **should** usually be `POST`, unless
-  the custom method maps more strongly to another HTTP verb.
-  - Custom operations that serve as an alternative to get or list operations
-    (such as `Search`) **should** use `GET`. These operations **must** be
-    idempotent and have no state changes or side effects (they should be safe
-    as defined in [RFC 7231][]).
-  - Custom operations **should not** use `PATCH` or `DELETE`.
-- The HTTP URI **must** use a `:` character followed by the custom verb
-  (`:archive` in the above example), and the verb in the URI **must** match the
-  verb in the name of the RPC.
-  - If word separation is required, `camelCase` **must** be used.
 - The `body` clause in the `google.api.http` annotation **should** be `"*"`.
   - However, if using `GET` or `DELETE`, the `body` clause **must** be absent.
-- Custom operations **should** usually take a request message matching the RPC
-  name, with a -`Request` suffix.
-- Custom operations **should** usually return a response message matching the
-  RPC name, with a -`Response` suffix.
+- Custom methods **should** usually take a request message matching the RPC
```

Comment:
why not must?

Reply:
Another issue:  "matching" is underspecified.

Can we make this guidance more precise, e.g. that it must be the RPC name + "Request"?

Or do we not care at all, because it's a custom method?

In general, it's nice for **should**s to have a counterexample.

Conclusion:  change this to **must** and clarify "matching"; we can always change it back to **should** if we find a good counterexample.

Reply:
Done.